app/src/main/java/org/kiwix/kiwixmobile/zimManager/libraryView/adapter/LibraryViewHolder.kt

```diff
@@ -60,21 +62,30 @@ sealed class LibraryViewHolder<in T : LibraryListItem>(containerView: View) :
       itemLibraryBinding.libraryBookLanguage.text = bookUtils.getLanguage(item.book.language)
       itemLibraryBinding.libraryBookFavicon.setBitmap(Base64String(item.book.favicon))
 
+      val hasAvailableSpaceInStorage = availableSpaceCalculator.hasAvailableSpaceForBook(item.book)
       containerView.setOnClickListener { clickAction.invoke(item) }
-      containerView.isClickable = item.canBeDownloaded
+      containerView.isClickable =
+        item.canBeDownloaded && hasAvailableSpaceInStorage
 
       itemLibraryBinding.tags.render(item.tags)
 
       itemLibraryBinding.unableToDownload.visibility =
-        if (item.canBeDownloaded) View.GONE else View.VISIBLE
+        if (item.canBeDownloaded && hasAvailableSpaceInStorage)
+          View.GONE
+        else
+          View.VISIBLE
       itemLibraryBinding.unableToDownload.setOnLongClickListener {
-        it.centreToast(
-          when (item.fileSystemState) {
-            CannotWrite4GbFile -> R.string.file_system_does_not_support_4gb
-            Unknown -> R.string.detecting_file_system
-            else -> throw RuntimeException("impossible invalid state: ${item.fileSystemState}")
+        when (item.fileSystemState) {
+          CannotWrite4GbFile -> it.centreToast(R.string.file_system_does_not_support_4gb)
+          Unknown -> it.centreToast(R.string.detecting_file_system)
+          else -> {
+            if (item.canBeDownloaded && !hasAvailableSpaceInStorage) {
+              clickAction.invoke(item)
```

Comment:
We should name these types more accurately. Here we don't know the difference between `Unknown` and `else`. We should add more accurate names.

Reply:
@gouri-panda , I believe that ```DetectingFileSystem``` would be a more accurate descriptor than ```Unknown```.

Reply:
@gouri-panda , is it looks good to you?

Reply:
I have changed ```Unknown``` to ```DetectingFileSystem``` and update test cases according to it.

Reply:
Looks good to me! Thanks!